internal/pkg/service/templates/api/dependencies/dependencies.go

```diff
@@ -149,13 +148,37 @@ func NewServerDeps(ctx context.Context, proc *servicectx.Process, cfg config.Con
 	}
 	logger.Infof("loaded Storage API index | %s", time.Since(startTime))
 
+	// Get timeout
+	// We use a longer timeout on the API start,
+	// the etcd could be started at the same time as the API.
+	connectTimeout := EtcdDefaultConnectionTimeout
+	if v, found := ctx.Value(EtcdConnectionTimeoutCtxKey).(time.Duration); found {
+		connectTimeout = v
+	}
+
+	// We use a longer timeout when starting the server, because etcd could be restarted at the same time as the API.
+	etcdCtx := context.WithValue(ctx, EtcdConnectionTimeoutCtxKey, 30*time.Second)
```

Comment:
Toto sa uz neaplikuje, ... to bol prave rozdiel pre optional etcd.
Ak sme sa nevedeli pripojit k etcd, tak sa to pri kazdom requeste, ktory potreboval etcd, skusalo znova.

No pri requeste sme nechceli cakat dlhsie ako `5s`, kedze etcd bolo optional.

------------

Teraz Templates API bez etcd nenastartuje, takze co sa tyka zavislosti, tak etcd klient bude vzdy dostupny.

Ak spojenie vypadne, tak retries si uz poriesit etcd klient interne.

-----

TLDR: malo by to byt rovnako ako v Buffer API

Reply:
Ahá, jasně. Nad tímhle jsem zrovna přemýšlel. 🙂 Fixnuto: https://github.com/keboola/keboola-as-code/pull/1279/commits/f8ce42ae63ed60c16d47e5dd6805cd02f86eb9cd